### PR TITLE
Adds URI#query_params method

### PR DESCRIPTION
**Pull request by rodrigopinto**
This PR is for adding a `URI#query_params` which will generate a `HTTP::Params` instance parsed from the `URI#query`.

Example:

```crystal
require "uri"

uri = URI.parse "http://foo.com?id=30&limit=5#time=1305298413"
uri.query_params # => HTTP::Params(@raw_params={"id" => ["30"], "limit" => ["5"]})
```
👉 it closes https://github.com/crystal-lang/crystal/issues/8085

### CODE REVIEW

File: spec/std/uri_spec.cr [hunk 1]
```diff
@@ -322,6 +322,21 @@ describe "URI" do
     end
   end
 
+  describe "#query_params" do
+    it do
+      uri = URI.parse("http://foo.com?id=30&limit=5#time=1305298413")
+      uri.query_params.should eq(HTTP::Params.parse("id=30&limit=5"))
```
Comment (straight-shoota) [inline spec/std/uri_spec.cr]: Technically, the spec would be more reliant if the `HTTP::Params` instance was created from a spec. We're not testing `Params.parse` here, but still...

Comment (rodrigopinto) [reply spec/std/uri_spec.cr]: What do you mean? Something like:

```crystal
expected_params = HTTP::Params.new({ "id" => ["30"], "limit" => ["5"]} )
uri.query_params.should eq(expected_params)
```

**Comment by straight-shoota** [comment]
@rodrigopinto Could you please squash the commits except the first one? I'd like to merge this as two commits, a spec refactor and a feature addition.

**Comment by rodrigopinto** [comment]
@straight-shoota it is done.

**Comment by straight-shoota** [comment]
Thank you @rodrigopinto

**Pull request merged**